admin_app_2/src/app/content/edit/page.tsx

```diff
@@ -0,0 +1,237 @@
+}
+
+interface EditContentBody {
+  content_title: string;
+  content_text: string;
+  content_language: string;
+  content_metadata: Record<string, unknown>;
+}
+
+const AddEditContentPage = () => {
+  const searchParams = useSearchParams();
+  const content_id = Number(searchParams.get("content_id")) || null;
+
+  const [content, setContent] = React.useState<Content | null>(null);
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+
+  React.useEffect(() => {
+    if (!content_id) {
+      setIsLoading(false);
+      return;
+    } else {
+      apiCalls.getContent(content_id).then((data) => {
+        setContent(data);
+        setIsLoading(false);
+      });
+    }
+  }, [content_id]);
+
+  if (isLoading) {
+    return (
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "row",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100vh",
+          width: "100%",
+        }}
+      >
+        <CircularProgress />
+      </div>
+    );
+  }
+  return (
+    <Layout.FlexBox flexDirection={"column"} sx={{ p: sizes.doubleBaseGap }}>
+      <Header content_id={content_id} />
+      <Layout.FlexBox
+        flexDirection={"column"}
+        sx={{ px: sizes.doubleBaseGap, mx: sizes.smallGap }}
+      >
+        <Layout.Spacer multiplier={2} />
+        <ContentBox content={content} setContent={setContent} />
+        <Layout.Spacer multiplier={1} />
+      </Layout.FlexBox>
+    </Layout.FlexBox>
+  );
+};
+
+const ContentBox = ({
+  content,
+  setContent,
+}: {
+  content: Content | null;
+  setContent: React.Dispatch<React.SetStateAction<Content | null>>;
+}) => {
+  const [isSaved, setIsSaved] = React.useState(true);
+  const [saveError, setSaveError] = React.useState(false);
+  const router = useRouter();
+  const saveContent = async (content: Content) => {
+    const body: EditContentBody = {
+      content_title: content.content_title,
+      content_text: content.content_text,
+      content_language: content.content_language,
+      content_metadata: content.content_metadata,
+    };
+
+    const promise =
+      content.content_id === null
+        ? apiCalls.addContent(body)
+        : apiCalls.editContent(content.content_id, body);
+
+    const result = promise
+      .then((data) => {
+        setIsSaved(true);
+        setSaveError(false);
+        return data.content_id;
+      })
+      .catch((error: Error) => {
+        console.error("Error processing content:", error);
+        setSaveError(true);
+        return null;
+      });
+
+    return await result;
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    key: keyof Content,
+  ) => {
+    const emptyContent: Content = {
+      content_id: null,
+      created_datetime_utc: null,
+      updated_datetime_utc: null,
+      content_title: "",
+      content_text: "",
+      content_language: "ENGLISH",
+      content_metadata: {},
+    };
```

Comment:
Change default to null instead of empty strings (as per your comment on my PR)?

Reply:
No this has a different role. This is when you are creating a new one and needs default values.

I already made the edit state one null:
`const [content, setContent] = React.useState<Content | null>(null);`